### PR TITLE
Feature/ldap authentication

### PR DESCRIPTION
**Pull request by jozefbakus**
Use LDAP to authenticate. (Issue: #41)
Merge two configs to one.
Add dialog to confirm workflow run via UI.
Refresh runs bug fix.

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala [hunk 1]
```diff
@@ -29,13 +29,14 @@ private object Configs {
...
-      } yield (key, value)).toMap
+      def getKeys(path: String): Seq[String] = {
+        Try(Configs.conf.getObject(path).keySet().asScala).getOrElse(Set.empty[String]) match {
+          case keys if keys.nonEmpty => keys.flatMap(k => getKeys(s"$path.$k")).toSeq
+          case keys if keys.isEmpty => Seq(path)
+        }
+      }
+      val keys = getKeys(propertyName)
+      keys.map(k => (k.stripPrefix(s"$propertyName."), conf.getString(k))).toMap
     }.getOrElse(Map.empty[String, String])
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala:40]: A test would be good for this non-trivial logic. Also, is it not possible to get all keys and then just filter them? It seems over-complicated to have recursion here. (Not your fault I guess. Seems like `typesafe.config` is not the simplest library out there)

File: pom.xml [hunk 2]
```diff
@@ -119,7 +120,6 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-tomcat</artifactId>
             <version>${spring-boot-starter-tomcat.version}</version>
-            <scope>provided</scope>
```
Comment (kevinwallimann) [inline pom.xml:122]: I don't understand this change. Why does the scope need to be `compile` now?

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
+import org.springframework.security.ldap.authentication.ad.ActiveDirectoryLdapAuthenticationProvider
+import org.springframework.stereotype.Component
+
+@Component
+class LdapAuthentication extends HyperdriverAuthentication {
+
+  @Value("${auth.ad.domain:}")
+  val adDomain: String = ""
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala]: Is the ` = ""` needed? You already specify a default value with the `:` in the `@Value`

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -5,4 +5,48 @@ logging.file=${logging.path}/app.log
 
 management.endpoints.web.base-path=/admin
 management.endpoints.jmx.exposure.exclude=*
-management.endpoints.web.exposure.include=*
\ No newline at end of file
+management.endpoints.web.exposure.include=*
```
Comment (Zejnilovic) [inline src/main/resources/application.properties]: 1. missing a new line
2. shouldn't you provide only template and not the actual properties?
3. Why is everything in auth commented out?
4. Provide more comments for the users as to what is what

File: src/main/resources/ui/controller/Workflows.controller.js [hunk 5]
```diff
@@ -77,6 +77,18 @@ sap.ui.define([
 			});
 		},
 
+		createConfirmRunWorkflowDialog: function (id) {
+			MessageBox.confirm("Are you sure you want to run workflow ?", {
```
Comment (Zejnilovic) [inline src/main/resources/ui/controller/Workflows.controller.js]:
```suggestion
			MessageBox.confirm("Are you sure you want to run workflow?", {
```

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala (hunk 3 above)
Comment (Zejnilovic) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala:27]: 1. In spring doc's there is always a newline between 2 props.
2. If the prop is missing, shouldn't there be an automatic error about it? Meaning you wouldn't have to validate.

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala [hunk 6]
```diff
@@ -29,13 +29,14 @@ private object Configs {
...
   def getMapFromConf(propertyName: String): Map[String, String] = {
     Try {
-      val list = Configs.conf.getObjectList(propertyName).asScala
-      (for {
-        item: ConfigObject <- list
-        entry <- item.entrySet().asScala
-        key = entry.getKey
-        value = entry.getValue.unwrapped().toString
-      } yield (key, value)).toMap
+      def getKeys(path: String): Seq[String] = {
```
Comment (Zejnilovic) [inline src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala:32]: Have you seen this [API DOC](https://lightbend.github.io/config/latest/api/com/typesafe/config/ConfigObject.html#unwrapped--)

File: pom.xml (hunk 2 above)
Comment (jozefbakus) [reply pom.xml:122]: It was your suggestion here https://github.com/AbsaOSS/hyperdrive/pull/10#discussion_r318432678
This dep. will be removed soon.

File: src/main/resources/application.properties (hunk 4 above)
Comment (jozefbakus) [reply src/main/resources/application.properties]: 1. Where are you missing a new line ? ... Removed/Fixed
2. True, management props are just for us (debugging) ... Removed/Fixed
3. Ok ... inmemory is uncommented now. ... Fixed
4. OK ... Fixed

File: src/main/resources/ui/controller/Workflows.controller.js (hunk 5 above)
Comment (jozefbakus) [reply src/main/resources/ui/controller/Workflows.controller.js]: Fixed.

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala (hunk 3 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala:27]: 1. Fixed. 
2. Yes, that is true, but if I leave it in spring it shows `Caused by: java.lang.IllegalArgumentException: Url cannot be empty` which is not that good as `Caused by: java.lang.IllegalArgumentException: auth.ad.domain has to be configured in order to use ldap authentication`.

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala (hunk 6 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala:32]: You are right. I have created issue #43 for that and it has to be properly handled.

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala (hunk 1 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala:40]: You are right. The same answer as for Sasa:  I have created issue #43 for that and it has to be properly handled.

File: src/main/resources/application.properties (hunk 4 above)
Comment (Zejnilovic) [reply src/main/resources/application.properties]: 1. End of the file

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala (hunk 3 above)
Comment (Zejnilovic) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala:27]: 2. Are you saying that `validateParams` will be called before the springs natural error?

Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/auth/LdapAuthentication.scala:27]: In general no, but in this case, a default value is provided.

File: src/main/resources/application.properties (hunk 4 above)
Comment (jozefbakus) [reply src/main/resources/application.properties]: 1. Done.

**Review by Zejnilovic — APPROVED** [review]
Thank you

**Comment by jozefbakus** [comment]
> Thank you

No, no, no ... THANK YOU!

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**